services/src/main/java/io/druid/cli/RouterJettyServerInitializer.java

```diff
@@ -84,6 +84,7 @@ public void initialize(Server server, Injector injector)
             )
         ), "/druid/v2/*"
     );
+    addExtensionFilters(root, injector);
```

Comment:
for this to work well with router nodes, filter implementations must be able to handle async servlet request propertly.
Maybe we should add a note in the ServletFilterHolder on that.